### DISCUSSION TITLE
Breadcrumbs navigation

### DISCUSSION THREAD

**Discussion by ziyaaktas** [vercel/next.js#15050] [Help]
What is a good starting point for building breadcrumbs navigation in nextjs.  Does the router support this sort of feature?

**Comment by cristiand391** [comment]
It should work, as long as you wrap all the links inside Next's `Link` component.
Are you using some library for breadcrumbs?

**Reply by ziyaaktas** [reply]
For UI, just material ui, but was actually wondering if nextjs has anything utilize for this.  
For example, for something like, url/parent/child/list, can I programmatically get these segments from nextjs?

**Reply by cristiand391** [reply]
You can use the [`useRouter`](https://nextjs.org/docs/api-reference/next/router#userouter) hook to get the actual page and use that value to dinamically style the current breadcrumb item.

**Reply by Pikachews** [reply]
What if your breadcrumbs don't depend on the route, but you want a breadcrumb if a component renders it?
e.g.
```
const TestPage = () => {
  return (
    <>
      <Breadcrumb>Test</Breadcrumb>
      <content></content> <----- may have breadcrumbs rendered under here.
    </>
  )
}
```
and your route is `/`, you don't want to build map from routes to breadcrumbs.

I've been looking into this library, but it doesn't worth with SSR: https://github.com/oklas/react-breadcrumbs-dynamic
but it does pretty much what I want.
